jctools-core/src/main/java/org/jctools/queues/MpmcArrayQueue.java

```diff
@@ -128,10 +128,13 @@ final boolean casConsumerIndex(long expect, long newValue)
  */
 public class MpmcArrayQueue<E> extends MpmcArrayQueueL3Pad<E>
 {
+    public static final int MAX_LOOK_AHEAD_STEP = Integer.getInteger("jctools.mpmc.max.lookahead.step", 4096);
```

Comment:
Should probably use `PortableJvmInfo.RECOMENDED_OFFER_BATCH` instead. On SPSC there's no reason not to claim large chunks for an uncontended producer, but for MPSC/MPMC the assumption should be some contention. Given that the empty slots in the queue will result in consumer 'bubbles' (queue not empty but slot is empty) we can potentially have an issue if a producer takes a while to fill up the slots it claims.